### PR TITLE
[Telemetry] Add S3 telemetry test

### PR DESCRIPTION
**Pull request by jeet4320**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ohadkatz** [comment]
LGTM

File: test/dlc_tests/sanity/test_telemetry.py [hunk 1]
```diff
@@ -289,11 +331,23 @@ def import_framework(image_uri, container_name, docker_cmd, framework, job_type,
...
+                warn=True
+            )
+            output = ec2_connection.run(
+                f"{docker_cmd} exec -i {container_name} /bin/bash -c 'cat /tmp/test_request.txt'",
+                warn=True
+            ).stdout.strip('\n')
+        else:
+            output = ec2_connection.run(
+                f"{docker_cmd} exec -i {container_name} python -c 'import {framework_to_import}; import time; time.sleep(5)'",
+                warn=True
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_telemetry.py:346]: why set warn to True here and handle error as assertion error?

File: test/dlc_tests/sanity/test_telemetry.py [hunk 2]
```diff
@@ -130,6 +131,38 @@ def test_pytorch_inference_job_type_env_var(pytorch_inference):
...
+
+    docker_cmd = "nvidia-docker" if processor == "gpu" else "docker"
+    test_utils.login_to_ecr_registry(ec2_connection, account_id, image_region)
+    ## For big images like trcomp, the ec2_connection.run command stops listening and the code hangs here.
+    ## Hence, avoiding the use of -q to let the connection remain active.
+    ec2_connection.run(f"{docker_cmd} pull {image_uri}")
+
+    actual_output = import_framework(image_uri, container_name, docker_cmd, framework, job_type, ec2_connection, test_mode = 1)
+
+    py_version = ec2_connection.run(
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_telemetry.py:153]: why do you need to get python version from container instead of image uri? i believe we already have a test that tests whether python version in the tag is equivalent to python version in the container.

File: test/dlc_tests/sanity/test_telemetry.py (hunk 1 above)
Comment (jeet4320) [reply test/dlc_tests/sanity/test_telemetry.py:346]: This is a copy paste of other code mentioned here https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/sanity/test_telemetry.py#L294, so thought to keep it same.

File: test/dlc_tests/sanity/test_telemetry.py (hunk 2 above)
Comment (jeet4320) [reply test/dlc_tests/sanity/test_telemetry.py:153]: that python version format is different than what we send for telemetry, for example telemetry needs 3.8.13, not py38 which comes from image uri

File: test/dlc_tests/sanity/test_telemetry.py [hunk 3]
```diff
@@ -62,6 +62,7 @@ def test_telemetry_instance_tag_failure_neuron(neuron, ec2_client, ec2_instance,
 def test_telemetry_instance_tag_success_gpu(gpu, ec2_client, ec2_instance, ec2_connection, non_huggingface_only, non_autogluon_only):
     _run_tag_success_IMDSv1(gpu, ec2_client, ec2_instance, ec2_connection)
     _run_tag_success_IMDSv2_hop_limit_2(gpu, ec2_client, ec2_instance, ec2_connection)
+    _run_s3_query_bucket_success(gpu, ec2_client, ec2_instance, ec2_connection)
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_telemetry.py:65]: Please create a separate test function for the S3 bucket query status.

This test function (all the `test_telemetry_instance_tag_*` tests) are meant to test the instance-tagging functionality.

File: test/dlc_tests/sanity/test_telemetry.py [hunk 4]
```diff
@@ -273,27 +309,44 @@ def _run_tag_success_IMDSv2_hop_limit_2(image_uri, ec2_client, ec2_instance, ec2
     assert expected_tag_key in ec2_instance_tags, f"{expected_tag_key} was not applied as an instance tag"
 
 
-def import_framework(image_uri, container_name, docker_cmd, framework, job_type, ec2_connection):
+def import_framework(image_uri, container_name, docker_cmd, framework, job_type, ec2_connection, test_mode = None):
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_telemetry.py:312]: Please change the name of this function to `trigger_telemetry` or something similar, because it handles the launching of a dummy TF inference service, as well as additional code now to return optionally the contents of the `/tmp/test_request.txt` file, if it is created.

**Review by saimidu — APPROVED** [review]

**Comment by saimidu** [comment]
Discussed offline - The error is a known issue, and will be handled in a separate PR dedicated to fixing the bug.

**Review by saimidu — DISMISSED** [review]

**Comment by saimidu** [comment]
Please remove the changes made to the pytorch/buildspec.yml file, and the conflicts should be resolved.

There is no need to approve again, after that. Please go ahead and merge from master into this branch, and then merge this PR with a squash+merge.

**Review by saimidu — APPROVED** [review]

**Review by saimidu — APPROVED** [review]
